orderer/common/multichannel/registrar.go

```
@@ -754,7 +796,12 @@ func (r *Registrar) removeMember(channelID string, cs *ChainSupport) error {
 func (r *Registrar) removeFollower(channelID string, follower *follower.Chain) error {
 	follower.Halt()
 
-	// TODO if follower is onboarding, remove the joinblock from file repo
+	_, status := follower.StatusReport()
+	if status == types.StatusOnBoarding {
+		if err := r.removeJoinBlock(channelID); err != nil {
```

That is not going to work if you have a follower that passed the join block (i.e. active) but did not yet remove the join block. That is especially true with the lazy removal that you implemented in this commit, but may also be true regardless because those things happen concurrently. 

So a better approach IMO would be to either (1) remove it unconditionally and ignore the error if it is not found (best to make a special error type to identify that) or (2) list join blocks and remove it only if it exists.

---

Changed this to always attempt removing the join block.